Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+            }
+
+            ProcessYAMLAssets(allFilesUnderAssets, Application.dataPath.Replace("Assets", "NuGet/Content"), remapDictionary, asmDefMappings);
+        }
+
+        private static Dictionary<string, string> RetrieveAsmDefGuids(string[] allFiles)
+        {
+            int lengthOfPrefix = Application.dataPath.IndexOf("Assets");
+            Dictionary<string, string> dllGuids = new Dictionary<string, string>();
+            foreach (string asmdefFile in allFiles.Where(t => t.EndsWith(".asmdef")))
+            {
+                string asmdefText = AssetDatabase.LoadAssetAtPath<AssemblyDefinitionAsset>(asmdefFile.Substring(lengthOfPrefix)).text;
+                string dllName = JsonUtility.FromJson<AssemblyDefinitionStub>(asmdefText).name;
+                string guid = File.ReadAllLines($"{asmdefFile}.meta")[1].Substring(6);
+                guid = CycleGuidForward(guid);
+                dllGuids.Add($"{dllName}.dll", guid);
+            }
+
+            return dllGuids;
+        }
+
+        private static void ProcessYAMLAssets(string[] allFilePaths, string outputDirectory, Dictionary<string, Tuple<string, long>> remapDictionary, Dictionary<string, string> dllGuids)
+        {
+            if (Directory.Exists(outputDirectory))
+            {
+                Directory.Delete(outputDirectory, true);
+            }
+
+            Debug.Log($"Output Directory: {outputDirectory}");
+
+            HashSet<string> foundNonYamlExtensions = new HashSet<string>();
+            List<Tuple<string, string>> yamlAssets = new List<Tuple<string, string>>();
+            foreach (string filePath in allFilePaths)
+            {
+                string targetPath = filePath.Replace(Application.dataPath, outputDirectory);
+                Directory.CreateDirectory(Path.GetDirectoryName(targetPath));
+
+                if (IsYamlFile(filePath))
+                {
+                    yamlAssets.Add(new Tuple<string, string>(filePath, targetPath));
+                }
+                else
+                {
+                    string extension = Path.GetExtension(filePath);
+                    if (!ExcludedYamlAssetExtensions.Contains(extension))
+                    {
+                        foundNonYamlExtensions.Add(extension);
+                    }
+
+                    bool copyFile = true;
+                    foreach (var suffix in ExcludedSuffixFromCopy)
+                    {
+                        if (filePath.EndsWith(suffix))
+                        {
+                            copyFile = false;
+                            break;
+                        }
+                    }
+
+                    if (copyFile)
+                    {
+                        File.Copy(filePath, targetPath);
+                    }
+                }
+            }
+
+            foreach (var extension in foundNonYamlExtensions)
+            {
+                Debug.Log($"Not a YAML extension: {extension}");
+            }
```

Comment:
Do you know if the point of this logging was mostly for development of this code, or does this debug output help with long term debuggability as well? It seems like given there's a .cs (and some other somewhat common extensions) this would probably log a bunch of lines.

Reply:
The way I have it setup is that I assume all assets are YAML but the ones in ExcludedYamlAssetExtensions; and then I double check by opening the file and looking for the header. If the header is missing, it isn't YAML and we should add that extension to ExcludedYamlAssetExtensions.

YAML can have arbitrary named extensions.

---
In reply to: [277830136](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277830136) [](ancestors = 277830136)